thumbor/engines/pil.py

```diff
@@ -452,50 +450,23 @@ def read(  # noqa
         return results
 
     def read_multiple(self, images, extension=None):
-        gif_writer = GifWriter()
-        img_buffer = BytesIO()
-
-        duration = []
-        converted_images = []
-        coordinates = []
-        dispose = []
-
-        for image in images:
-            duration.append(image.info.get("duration", 80) / 1000)
-            converted_images.append(image.convert("RGB"))
-            coordinates.append((0, 0))
-            dispose.append(1)
-
-        loop = int(self.image.info.get("loop", 1))
-
-        images = gif_writer.convertImagesToPIL(converted_images, False, None)
-        gif_writer.writeGifToFile(
-            img_buffer, images, duration, loop, coordinates, dispose
-        )
-
-        results = img_buffer.getvalue()
-        img_buffer.close()
-
-        tmp_fd, tmp_file_path = mkstemp()
-        temp_file = os.fdopen(tmp_fd, "wb")
-        temp_file.write(results)
-        temp_file.close()
-
-        command = ["gifsicle", "--colors", "256", tmp_file_path]
-
-        popen = Popen(  # pylint: disable=consider-using-with
-            command, stdout=PIPE
-        )
-        pipe = popen.stdout
-        pipe_output = pipe.read()
-        pipe.close()
+        with NamedTemporaryFile() as tmp_file:
+            images[0].save(
+                tmp_file.name,
+                format="GIF",
+                save_all=True,
+                append_images=images[1:],
+                duration=[im.info.get("duration", 80) / 1000 for im in images],
+                loop=int(self.image.info.get("loop", 1)),
+            )
 
-        if popen.wait() == 0:
-            results = pipe_output
+            command = ["gifsicle", "--colors", "256", tmp_file.name]
```

Comment:
It seems that Pillow improved Gif support, we might not need gifsicle anymore https://github.com/search?q=repo%3Apython-pillow%2FPillow+gif+language%3Arst+path%3Adocs&type=code

Reply:
Yes! I'm coming to the exact same conclusion. After some experimentation, I'm seeing hardly any improvements to GIFs with `gifsicle`, to a point where it doesn't make much sense to keep supporting it, especially by running it as a sub process.